File: Cocoa Programming/Cocoa Pr L63 ObjC/Cocoa Pr L63 ObjC/CircleView.m
//
//  CircleView.m
//  Cocoa Pr L63 ObjC
//
//  Created by Michele Galvagno on 10/06/22.
//

#import "CircleView.h"

@implementation CircleView

- (BOOL)wantsUpdateLayer {
    return YES;
}

- (void)updateLayer {
    self.layer.backgroundColor = NSColor.whiteColor.CGColor;
    self.layer.cornerRadius = (self.frame.size.width)/2.0;
}

- (void)removeCircle:(NSView *)circle {
    [circle removeFromSuperview];
}

@end

File: Cocoa Programming/Cocoa Pr L63 NSUndoManager/Cocoa Pr L63 NSUndoManager/CircleView.swift
//
//  CircleView.swift
//  Cocoa Pr L63 NSUndoManager
//
//  Created by Michele Galvagno on 09/06/22.
//

import Cocoa

class CircleView: NSView {

    override var wantsUpdateLayer: Bool {
        return true
    }
    
    override func updateLayer() {
        layer?.backgroundColor = NSColor.white.cgColor
        layer?.cornerRadius = frame.size.width/2.0
    }
    
}
